### PR TITLE
replace unmaintainable jsqrcode

### PR DESCRIPTION
**Pull request by FantasqueX**
The motivation is to remove unmaintainable jsqrcode.

First, I try to use zxing/browser as an alternative. However, it uses zxing/library as an implicit dependent which I think is dangerous. It is also not under active development.

Why to use nuintun/qrcode? It is active currently with detailed document.

I'm not sure whether this is the best alternative. Any suggestion is welcome.

### CODE REVIEW

**Comment by mzz2017** [comment]
@FantasqueX nuintun/qrcode is good. Actually I do not care about which lib to use but I care about if it works fine because it is not a core component that impacts the performance and user expererence.

**Comment by FantasqueX** [comment]
I have tested it locally. Because of lacking unit test, it needs to be reviewed.

I know it is not a core component, maybe not many people use this feature. However, we shouldn't use dead projects as dependency. Code quality is important as well.

File: gui/src/node.vue [hunk 1]
```diff
@@ -692,7 +692,7 @@
 import { locateServer, handleResponse } from "@/assets/js/utils";
 import CONST from "@/assets/js/const";
 import QRCode from "qrcode";
```
Comment (mzz2017) [inline gui/src/node.vue:694]: Can we also replace this with `@nuintun/qrcode`?

Comment (FantasqueX) [reply gui/src/node.vue:694]: I think it is possible. Let me fix it.

Comment (FantasqueX) [reply gui/src/node.vue]: Done.

Comment (mzz2017) [reply gui/src/node.vue]: I see the QR code is not center aligned, can you fix it?

Comment (mzz2017) [reply gui/src/node.vue]: And it occupys more unnecessary width.
image

Comment (mzz2017) [reply gui/src/node.vue]: Sometimes, it has incorrect ratio.
image

Comment (FantasqueX) [reply gui/src/node.vue]: Sorry for lacking tests. I don't think it is appropriate to use @nuintun/qrcode to replace the current node-qrcode. Nuintun's one can only export a non-scalable SVG format qrcode image. I think I should revert the last commit and keep both libraryies.

Comment (mzz2017) [reply gui/src/node.vue:694]: It seems img works fine. Code:

```diff
diff --git a/gui/src/node.vue b/gui/src/node.vue
index aff9e9b..a26b752 100644
--- a/gui/src/node.vue
+++ b/gui/src/node.vue
@@ -1436,7 +1436,7 @@ export default {
                         }</p>
                     </header>
                     <section class="modal-card-body lazy" style="text-align: center">
-                        <div><canvas id="canvas" class="qrcode"></canvas></div>
+                        <div><img id="qrcode" src=""/></div>
                         <div class="tags has-addons is-centered" style="position: relative">
                             <span class="tag is-rounded is-dark sharingAddressTag" style="position: relative" data-clipboard-text="${
                               res.data.data.sharingAddress
@@ -1474,20 +1474,15 @@ export default {
             if (row._type === CONST.SubscriptionType) {
               add = "sub://" + Base64.encode(add);
             }
-            const qrcode = new Encoder();
+            const enc = new Encoder();
 
-            qrcode.setEncodingHint(true);
-            qrcode.setErrorCorrectionLevel(ErrorCorrectionLevel.H);
-            qrcode.write(add);
-            qrcode.make();
+            enc.setEncodingHint(true);
+            enc.setErrorCorrectionLevel(ErrorCorrectionLevel.H);
+            enc.write(add);
+            enc.make();
 
-            const canvas = document.getElementById("canvas");
-            const context = canvas.getContext("2d");
-            const img = new Image();
-            img.onload = function() {
-              context.drawImage(img, 0, 0); // Or at whatever offset you like
-            };
-            img.src = qrcode.toDataURL();
+            const qrcode = document.getElementById("qrcode");
+            qrcode.src = enc.toDataURL(80 / Math.sqrt(add.length));
 
             let targets = document.querySelectorAll(".sharingAddressTag");
             let covers = document.querySelectorAll(".tag-cover");
@@ -1738,7 +1733,7 @@ table th {
   font-size: 40px;
 }
 
-.qrcode#canvas {
+#qrcode {
   min-height: 300px !important;
   min-width: 300px !important;
 }
```

**Pull request merged**